HIP/hip-xxx.md

```diff
@@ -0,0 +1,921 @@
+                    "0000000000000000000000000000000000000000000000000000000000000014",
+                    "0000000000000000000000000000000000000000000000000000000000000014"
+                ],
+                "memory": [
+                    "0000000000000000000000000000000000000000000000000000000000000000",
+                    "0000000000000000000000000000000000000000000000000000000000000000",
+                    "0000000000000000000000000000000000000000000000000000000000000080"
+                ],
+                "storage": null,
+                "reason": null
+            },
+            [...],
+        ]
+    }
+}           
+```
+
+### Mirror node Trace REST API
+
+The JSON-RPC Relay will in turn call the Mirror Node REST API. Depending on the `tracer` that was provided as an input parameter to the `debug_traceTransaction` RPC endpoint, one of two Mirror node endpoints will be called:
+1. For `debug_traceTransaction` called with `callTracer`, the Mirror Node `/api/v1/contracts/results/{transactionIdOrHash}/actions` endpoint will be called.
+2. For `debug_traceTransaction` called with `opcodeLogger`, the Mirror Node `/api/v1/contracts/results/{transactionIdOrHash}/opcodes` endpoint will be called.
+
+The `/actions` endpoint is already implemented. 
+
+This section will specify the `/opcodes` endpoint.
+
+#### `/api/v1/contracts/results/{transactionIdOrHash}/opcodes`
+
+**Input parameters:** The endpoint will accept three optional parameters:
+
+```
+// Boolean flag controlling whether to return stack information. Defaults to `true`.
+
+stack: <bool> 
+
+// Boolean flag controlling whether to return memory information. Defaults to `false`.
+
+memory: <bool> 
+
+// Boolean flag controlling whether to return storage information. Defaults to `false`.
+
+storage: <bool>
+```
+
+Disabling stack, memory and storage traces will significantly reduce the size of the opcode related data. This would be helpful for reducing the network bandwidth. 
+
+**Output**: detailed information for the transaction execution with `stack`, `memory` and `storage` set to `true`
+Example output:
+
+```
+{
+    "gas": 52139,
+    "failed": false,
+    "returnValue": "0x0000000000000000000000000000000000000000000000000000000000000001",
+    "structLogs": [
+        [...],
+        {
+            "pc": 542,
+            "op": "SWAP2",
+            "gas": 4419,
+            "gasCost": 3,
+            "depth": 1,
+            "stack": [
+                "0x00000000000000000000000000000000000000000000000000000000094f6253",
+                "00000000000000000000000000000000000000000000000000000000000000a5",
+                "0000000000000000000000008fa74f0a79112c8c46c8e6739eb8cc929083298c",
+                "0000000000000000000000000000000000000000000000000000000000000000",
+                "0000000000000000000000000000000000000000000000000000000000000014",
+                "0000000000000000000000000000000000000000000000000000000000000014"
```

Comment:
Not all binary data has `0x` prefix here and below

Reply:
Added

Reply:
Still some `0x` missing in first example